### PR TITLE
Ormolu flags

### LINKED ISSUE #245: Ormolu formatting fails with CPP pragma
For any module with a CPP language pragma, the Ormolu formatter fails with:
```
2020-07-27 20:25:20.119291 [ThreadId 67857] - finish: Ormolu (took 0.00s)
2020-07-27 20:25:20.120322 [ThreadId 7] - <--2--{"error":{"code":-32602,"message":"ormoluCmd: OrmoluUnrecognizedOpts (\"-XCpp\" :| [])"},"jsonrpc":"2.0","id":825}
[Error - 8:25:20 PM] Request textDocument/formatting failed.
  Message: ormoluCmd: OrmoluUnrecognizedOpts ("-XCpp" :| [])
  Code: -32602 
```

Command line Ormolu succeeds, so this is an HLS bug

### PR DESCRIPTION
**Pull request by pepeiborra**
There's two independent changes here:

1. Fix the rendering of the CPP extension as a flag; there might be a better way to fix this.
2. Do not process LANGUAGE and OPTIONS pragmas for Ormolu, since I believe Ormolu is going to process those anyway. @mrkkrp is that right?

Fixes #245

### CODE REVIEW

File: src/Ide/Plugin/Ormolu.hs [hunk 1]
```diff
@@ -36,24 +43,24 @@ descriptor plId = (defaultPluginDescriptor plId)
 -- ---------------------------------------------------------------------
 
 provider :: FormattingProvider IO
-provider _lf ideState typ contents fp _ = do
+provider lf ideState typ contents fp _ = withIndefiniteProgress lf title Cancellable $ do
```
Comment (fendor) [inline src/Ide/Plugin/Ormolu.hs:46]: I guess we should have these for all formatting providers?

Comment (pepeiborra) [reply src/Ide/Plugin/Ormolu.hs:46]: In many other plugins too, specially in the Eval plugin

Comment (bubba) [inline src/Ide/Plugin/Ormolu.hs:46]: Can we cancel operations in ghcide/hls? Should it be `NotCancellable` here

**Comment by mrkkrp** [comment]
@pepeiborra Depends on what you mean by "process".

File: src/Ide/Plugin/Ormolu.hs (hunk 1 above)
Comment (pepeiborra) [reply src/Ide/Plugin/Ormolu.hs:46]: ghcide does implement cancellations, so I think this should be `Cancellable` in general

**Comment by pepeiborra** [comment]
> @pepeiborra Depends on what you mean by "process".

Does Ormolu use ghc to parse source files? If yes, then ghc will see the language and options pragmas and act accordingly

**Comment by mrkkrp** [comment]
@pepeiborra In that sense, yes, we use functions from GHC to read pragmas before we run the parser.

**Comment by fendor** [comment]
So, this is ready for merge, right?

**Pull request merged**